File: 2364-count-number-of-bad-pairs/2364-count-number-of-bad-pairs.py
class Solution:
    def countBadPairs(self, nums: List[int]) -> int:
      
      length = len(nums)
      
      for j in range(length):
        nums[j] -= j
      
      # print(nums)
      
      Dict = defaultdict(int)
      
      count = 0
      
      for i in range(length):
        count += Dict[nums[i]]
        Dict[nums[i]] += 1
      
      return int((length*(length-1))/2 - count)
      
        

File: 2364-count-number-of-bad-pairs/2364-count-number-of-bad-pairs.cpp
class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
      
      long long count = 0;
      
      for (int i = 0; i < nums.size(); i++)
        nums[i] -= i;
      
      unordered_map<int,vector<int>> map1;
      // vector<int> dp(nums.size(),0);
      
      for (int i = 0; i < nums.size(); i++)
      {
        count += map1[nums[i]].size();
        map1[nums[i]].push_back(i);
      }
      return (long long)nums.size()*(nums.size()-1)/2 - count;
    }
};